src/main/scala/viper/gobra/frontend/PackageResolver.scala

```diff
@@ -124,13 +166,140 @@ object PackageResolver {
 
   private lazy val pkgClauseRegex = """(?:\/\/.*|\/\*(?:.|\n)*\*\/|package(?:\s|\n)+([a-zA-Z_][a-zA-Z0-9_]*))""".r
 
-  private def getPackageClause(file: File): Option[String] = {
-    val bufferedSource = scala.io.Source.fromFile(file)
+  private def getPackageClause(file: InputResource): Option[String] = {
+    val inputStream = file.asStream()
+    val bufferedSource = new BufferedSource(inputStream)
     val content = bufferedSource.mkString
     bufferedSource.close()
+
     // TODO is there a way to perform the regex lazily on the file's content?
     pkgClauseRegex
       .findAllMatchIn(content)
       .collectFirst { case m if m.group(1) != null => m.group(1) }
   }
+
+  trait InputResource extends Closeable {
+    val path: Path
+
+    def resolve(pathComponent: String): InputResource
+
+    def exists(): Boolean = Files.exists(path)
+
+    private var stream: Option[InputStream] = None
+    /**
+      * stream has to be closed after use either by calling `close` on the resource or directly on the stream.
+      * In case the resource (if applicable incl. its file system) is continued to be used by the stream is no
+      * longer needed, it is recommended to directly call `close` on the stream as soon as possible and call `close`
+      * on the resource when it is no longer needed.
+      */
+    def asStream(): InputStream = {
+      val inputStream = Files.newInputStream(path)
+      stream = Some(inputStream)
+      inputStream
+    }
+
+    /** returns files that are part of this directory */
+    def listContent(): Vector[InputResource]
+
+    /** closes stream and filesystem (if applicable) */
+    override def close(): Unit = stream match {
+      case Some(s) => {
+        s.close()
+        stream = None
+      }
+      case _ =>
+    }
+  }
+
+  case class FileResource(file: File) extends InputResource {
+    override lazy val path: Path = file.toPath
+
+    def resolve(pathComponent: String): FileResource =
```

Comment:
isn't the `override` keyword missing for this declaration?

Reply:
It is. IntelliJ does not even issue a warning for this